sql-plugin/src/main/scala/org/apache/spark/sql/rapids/misc.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids
+
+import ai.rapids.cudf.ColumnVector
+import com.nvidia.spark.rapids.{GpuColumnVector, GpuUnaryExpression}
+
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression}
+import org.apache.spark.sql.types.{AbstractDataType, DataType, NullType, StringType}
+
+case class GpuRaiseError(child: Expression) extends GpuUnaryExpression with ExpectsInputTypes {
+
+  override def dataType: DataType = NullType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType)
+  override def toString: String = s"raise_error($child)"
+
+  /** Could evaluating this expression cause side-effects, such as throwing an exception? */
+  override def hasSideEffects: Boolean = true
+
+  override protected def doColumnar(input: GpuColumnVector): ColumnVector = {
+    if (input == null || input.getRowCount <= 0) {
+      throw new RuntimeException()
+    }
+
+    // Take the first one as the error message
+    val msg = input.copyToHost().getUTF8String(0).toString
```

Comment:
Is there any necessary to only copy the first row to get the error msg instead of copying the whole column vector?

Reply:
This is on an error case. The entire job is going to fail. I am not too concerned with failing faster than the CPU. Yes it would be nice to not copy everything. You can do that with `getScalarElement`, which should keep the code small and clean.

```
withResource(input.getScalarElement(0)) { scalarMsg =>
  if (!scalarMsg.isValid()) {
    throw new RuntimeException()
  } else {
    throw new RuntimeException(scalarMsg.getJavaString())
  }
}
```

Reply:
Good suggestion. Done.